pkg/api/service/warnings.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"fmt"
+	"net/netip"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/apis/core/helper"
+)
+
+func GetWarningsForService(service, oldService *api.Service) []string {
+	if service == nil {
+		return nil
+	}
+	var warnings []string
+
+	if helper.IsServiceIPSet(service) {
+		warnings = append(warnings, getWarningsForClusterIPs(service)...)
+	}
+
+	if service.Spec.Type == api.ServiceTypeExternalName {
+		warnings = append(warnings, getWarningsForExternalName(service)...)
+	}
+
+	if oldService != nil {
+		warnings = append(warnings, getWarningsForServiceTransitions(service, oldService)...)
+	}
+
+	return warnings
+}
+
+// getWarningsForClusterIPs returns warnings for ClusterIPs
+func getWarningsForClusterIPs(service *api.Service) []string {
+	var warnings []string
+	var fieldPath *field.Path
+
+	for i, clusterIP := range service.Spec.ClusterIPs {
+		// IPs with leading zeros CVE-2021-29923 are not valid in golang since 1.17
```

Comment:
This problem applies to ExternalIPs, LoadBalancerIP, and LoadBalancerSourceRanges too.

Reply:
yep yep yep